src/proxy/ListProxy.ts

```diff
@@ -132,15 +132,15 @@ export class ListProxy<E> extends PartitionSpecificProxy implements IList<E> {
         const listenerHandler = (message: ClientMessage) => {
             ListAddListenerCodec.handle(message, (element: Data, uuid: string, eventType: number) => {
                 const responseObject = element ? this.toObject(element) : null;
-                let listenerFunction: Function;
-                if (eventType === 1) {
-                    listenerFunction = listener.itemAdded;
-                } else if (eventType === 2) {
-                    listenerFunction = listener.itemRemoved;
-                }
 
-                if (listenerFunction) {
-                    listenerFunction.call(responseObject, responseObject);
+                const member = this.client.getClusterService().getMember(uuid);
```

Comment:
Minor comment: Looks like this part is repeated. Is it possible to re-use this part in different proxies?

Reply:
I think it is not a redundancy. It accesses the member with just 2 hops.